state convert blog (#3428)
pulumi/pulumi-hugo: themes/default/content/blog/converting-full-terraform-states-to-pulumi/index.md

---
title: "Converting Full Terraform States to Pulumi"

> date: 2023-06-12

Reviewer: Incorrect date?

Author: Yeh copy paste

meta_desc: Learn how to convert whole Terraform states to Pulumi using the new Terraform converter
meta_image: meta.png
authors:
    - fraser-waters
tags:
    - migration
---

> Building on our work of [converting Terraform projects](/docs/converting-full-terraform-programs-to-pulumi), we now have support for converting Terraform state.

Reviewer: ```suggestion
Building on our work of [converting Terraform projects](/docs/converting-full-terraform-programs-to-pulumi), we now have support for adopting resources from Terraform state
```

<!--more-->

> In the prior blog post about converting from Terraform we called out that while it was great that the converter could migrate a Terraform project, but what if you had existing infrastructure resources.

Reviewer 2: ```suggestion
In the prior blog post about converting from Terraform we called out that it was great that the converter could migrate a Terraform project, but it didn't address situations where you had existing infrastructure resources.
```

Historically we've shared some AWS specific code for [TypeScript](https://github.com/pulumi/tf2pulumi/blob/master/misc/import/import.ts) and [Go](https://github.com/pulumi/tf2pulumi/blob/master/misc/import-go/import.go) to import from `.tfstate` files. These have been useful, but they're limited to the AWS provider and either Go or TypeScript programs.

> From [v3.87.0](/docs/install) we have extended our converter system to state not just code. So now it's possible to import resources from other state system, the first of which we have support for being being Terraform again.

Reviewer 2: ```suggestion
From [v3.87.0](/docs/install) we have extended our converter system to also include state in addition to code. So now it's possible to import resources from other state systems, the first of which is Terraform.
```
